Fabric/src/main/java/me/ferdz/placeableitems/block/component/impl/FragileBlockComponent.java

```diff
@@ -0,0 +1,97 @@
+package me.ferdz.placeableitems.block.component.impl;
+
+import java.util.function.Supplier;
+
+import me.ferdz.placeableitems.block.component.AbstractBlockComponent;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.Hand;
+import net.minecraft.util.hit.BlockHitResult;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+public class FragileBlockComponent extends AbstractBlockComponent {
+
+    private ContextualObjectCreator<ItemStack> itemStackCreator;
+    private ContextualObjectCreator<? extends Entity> entityCreator;
+
+    public FragileBlockComponent(ContextualObjectCreator<ItemStack> itemStackCreator, ContextualObjectCreator<? extends Entity> entityCreator) {
+        this.itemStackCreator = itemStackCreator;
+        this.entityCreator = entityCreator;
+    }
+
+    public FragileBlockComponent(Supplier<ItemStack> itemStackCreator, Supplier<? extends Entity> entityCreator) {
+        this(ContextualObjectCreator.fromSupplier(itemStackCreator), ContextualObjectCreator.fromSupplier(entityCreator));
+    }
+
+    public FragileBlockComponent(ContextualObjectCreator<ItemStack> itemStackCreator, Supplier<? extends Entity> entityCreator) {
+        this(itemStackCreator, ContextualObjectCreator.fromSupplier(entityCreator));
+    }
+
+    public FragileBlockComponent(Supplier<ItemStack> itemStackCreator, ContextualObjectCreator<? extends Entity> entityCreator) {
+        this(ContextualObjectCreator.fromSupplier(itemStackCreator), entityCreator);
+    }
+
+    public FragileBlockComponent() {
+        this((ContextualObjectCreator<ItemStack>) null, (ContextualObjectCreator<? extends Entity>) null);
+    }
```

Comment:
This is a lot of unused initializers and unused `with...` chainings, please keep it as simple as possible. We should only keep the single best way of doing these things to remove as much overhead / confusion / maintenance problems as possible